browser/components/newtab/aboutNewTabService.js

```diff
@@ -139,15 +153,16 @@ AboutNewTabService.prototype = {
   },
 
   /*
-   * Generate a default url based on locale and update channel
+   * Generate a default url based on remote mode, version, locale and update channel
    */
   generateRemoteURL() {
     let releaseName = this.releaseFromUpdateChannel(UpdateUtils.UpdateChannel);
-    let url = REMOTE_NEWTAB_URL
+    let path = REMOTE_NEWTAB_PATH
       .replace("%VERSION%", REMOTE_NEWTAB_VERSION)
       .replace("%LOCALE%", Locale.getLocale())
       .replace("%CHANNEL%", releaseName);
-    return url;
+    let mode = Services.prefs.getCharPref(PREF_REMOTE_MODE, "production");
+    return MODE_CHANNEL_MAP[mode].origin + path;
```

Comment:
We need to watch out for invalid `mode` arguments. The default fallback should be `production`.